pxd.c

```diff
@@ -1521,10 +1521,10 @@ ssize_t pxd_add(struct fuse_conn *fc, struct pxd_add_ext_out *add)
 		pxd_dev->queue_depth = add->queue_depth;
 	}
 
-	if (add->discard_size < SECTOR_SIZE)
-		pxd_dev->discard_size = SEGMENT_SIZE;
+	if (add->discard_size <= PXD_MAX_DISCARD_GRANULARITY)
+		pxd_dev->discard_size = PXD_MAX_DISCARD_GRANULARITY;
 	else
-		pxd_dev->discard_size = add->discard_size;
+		pxd_dev->discard_size = ALIGN(add->discard_size, PXD_MAX_DISCARD_GRANULARITY);
```

Comment:
if discard disabling is to be supported, then  this portion should support discard_size as zero too, instead of upping it to discard granularity.